src/common/blkdev.cc

```diff
@@ -1208,3 +1266,36 @@ int block_device_run_nvme(const char *device, const char *vendor, int timeout,
 }
 
 #endif
+
+
+
+void get_device_metadata(
+  const std::set<std::string>& devnames,
+  std::map<std::string,std::string> *pm,
+  std::map<std::string,std::string> *errs)
+{
+  (*pm)["devices"] = stringify(devnames);
+  string &devids = (*pm)["device_ids"];
+  string &devpaths = (*pm)["device_paths"];
+  for (auto& dev : devnames) {
+    string err;
+    string id = get_device_id(dev, &err);
+    if (id.size()) {
+      if (!devids.empty()) {
+	devids += ",";
+      }
+      devids += dev + "=" + id;
+    } else {
+      (*errs)[dev] = " no unique device id for "s + dev + ": " + err;
+    }
+    string path = get_device_path(dev, &err);
```

Comment:
I'm trying to converge to `device_id`s. E.g. in #31072 . That's not enough?

Reply:
I think that's good for the user-facing part.  But lsmcli needs a path name, and I'm running into this issue where say osd.3 is /dev/sdm, i make it blink, pull the drive, put a new one in, and it gets assigned /dev/sdx.  To turn the light off i'm currently manually tailing dmesg to get the new device name.  Using the path-based variants allow the stale path to remain valid.

(It's still a bit awkward tho in that you can't turn the light off in my chassis if there isn't a drive in the slot.  But I blame lsmcli for that.)

Reply:
Reminds me, where the dashboard uses device file names to match `osd_metadata` and the potentially outdated orch inventory to map OSDs to devices:

https://github.com/ceph/ceph/blob/0f68bec9759d476d2edd3ea574ba8340c4364464/src/pybind/mgr/dashboard/controllers/orchestrator.py#L23-L54